src/monodroid/jni/dylib-mono.h

```diff
@@ -295,10 +269,10 @@ struct DylibMono {
 	monodroid_mono_method_full_name_fptr                    mono_method_full_name;
 	monodroid_mono_object_get_class_fptr                    mono_object_get_class;
 	monodroid_mono_object_unbox_fptr                        mono_object_unbox;
-	monodroid_mono_profiler_install_fptr                    mono_profiler_install;
-	monodroid_mono_profiler_install_jit_end_fptr            mono_profiler_install_jit_end;
-	monodroid_mono_profiler_install_thread_fptr             mono_profiler_install_thread;
-	monodroid_mono_profiler_set_events_fptr                 mono_profiler_set_events;
+	monodroid_mono_profiler_install_fptr                     mono_profiler_install;
+	monodroid_mono_profiler_set_jit_done_callback_fptr       mono_profiler_set_jit_done_callback;
```

Comment:
This is [potentially problematic](https://github.com/xamarin/xamarin-android/blob/15f7547/src/monodroid/jni/dylib-mono.h#L256):

    /* NOTE: structure members MUST NOT CHANGE ORDER. */

The reason for that comment is because a long time ago @mhutch wanted to reuse `dylib-mono.h` for use with [Unreal](https://mono-ue.github.io). Changing *any* members would constitute an ABI breakage; it was expected that any callers could check the `DylibMono::version` structure to see if it was >= the size they expected, and bail if it was "too small."

This only works if we only append structure members, i.e. the "version number" is constantly increasing, hence the comment.

This PR changes member ordering, which could break any external users of `monodroid_dylib_mono_new()`...if there were any.

*Which there probably aren't*.

I'm thus conflict: if there *were* or *will be* external users of `monodroid_dylib_mono_new()`, then this particular change in this PR isn't acceptable. Which could be [very problematic if the "obsolete" symbols are ever *removed* from mono](https://github.com/xamarin/xamarin-android/blob/15f7547/src/monodroid/jni/dylib-mono.c#L142-L145).

Pinging @mhutch to confirm that we don't care about ABI here. If we do need to care, we'll likely need to reconsider the public interface to better deal with "optional" `DylibMono` sections.